swarm/src/connection/pool/concurrent_dial.rs

```diff
@@ -153,12 +154,12 @@ fn p2p_addr(peer: Option<PeerId>, addr: Multiaddr) -> Result<Multiaddr, Multiadd
         None => return Ok(addr),
     };
 
-    if let Some(multiaddr::Protocol::P2p(hash)) = addr.iter().last() {
+    if let Some(Protocol::P2p(hash)) = addr.iter().last() {
```

Comment:
Can't comment on the file directly so will do it here:

Why can't `ConcurrentDial` not stay within `libp2p-core`? It only depends on `libp2p-core` defined interfaces from what I can see so could be a useful primitive for building on top of `libp2p-core` without `libp2p-swarm`.